### PR TITLE
Support ghc-9.2.3

### PR DESCRIPTION
**Pull request by July541**
Dropped ghc-8.10.6 and ghc-9.0.1, keep supporting ghc-9.2.2.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2936"></a>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -98,12 +97,10 @@ jobs:
...
              test: true
            # only build rest of supported ghc versions for windows
-           - os: windows-latest
-             ghc: '9.0.1'
-           - os: windows-latest
-             ghc: '8.10.6'
            - os: windows-latest
              ghc: '8.8.4'
+           - os: windows-latest
+             ghc: '9.2.2'
```
Comment (pepeiborra) [inline .github/workflows/test.yml:103]: Why not drop "9.2.2" here as well?

File: cabal.project [hunk 2]
```diff
@@ -67,26 +67,5 @@ allow-newer:
...
-
-  -- for ekg
-  ekg-core:base,
-  ekg-core:ghc-prim,
-  ekg-wai:base,
-  ekg-wai:time,
-
-  -- for shake-bench
-  Chart-diagrams:diagrams-core,
-  SVGFonts:diagrams-core
```
Comment (pepeiborra) [inline cabal.project:92]: awesome!

**Review by pepeiborra — APPROVED** [review]
Thank you so much! Just wondering whether we can drop `9.2.2` altogether

**Comment by July541** [comment]
> Thank you so much! Just wondering whether we can drop `9.2.2` altogether

Following https://haskell-language-server.readthedocs.io/en/latest/supported-versions.html#minor-versions, it says "For the latest supported major GHC version we will support at least 2 minor versions". I think the "2 minor versions" means 9.2.2 and 9.2.3.

**Comment by July541** [comment]
stack build failed because hackage doesn't have ghc-9.2.3 yet, let's wait for it.

**Comment by michaelpj** [comment]
Great stuff!

> stack build failed because hackage doesn't have ghc-9.2.3 yet, let's wait for it.

Might need to prod the GHC devs about this, they sometimes forget to upload it.

**Comment by July541** [comment]
> Might need to prod the GHC devs about this, they sometimes forget to upload it.

Done. https://gitlab.haskell.org/ghc/ghc/-/issues/21326#note_434846

**Comment by mouse07410** [comment]
Nightly for GHC-9.2.3 is already on Stackage. 

Is this PR ready to be merged?

**Comment by July541** [comment]
We should merge #2944 first because build with ghc-9.2.3 will choose fourmolu-0.7.0.0.

**Comment by michaelpj** [comment]
`fourmolu` tests fail, I guess maybe https://github.com/haskell/haskell-language-server/pull/2944 didn't have a test run where we pick then new version? cc @georgefst

**Comment by georgefst** [comment]
> `fourmolu` tests fail, I guess maybe #2944 didn't have a test run where we pick then new version? cc @georgefst

Hopefully fixed in #2950.

**Comment by georgefst** [comment]
> > `fourmolu` tests fail, I guess maybe #2944 didn't have a test run where we pick then new version? cc @georgefst
> 
> Hopefully fixed in #2950.

It's merged! Hopefully this PR is now good to go.

**Comment by mouse07410** [comment]
Great! Time for a new release? 😉

**Comment by July541** [comment]
> Great! Time for a new release? 😉

As you may saw, hls is trying some improvements recently, and it may be worth waiting until they are ready :)

**Pull request merged**